_ml-commons-plugin/agents-tools/tools/ppl-tool.md

```diff
@@ -0,0 +1,192 @@
+---
+layout: default
+title: PPL tool
+has_children: false
+has_toc: false
+nav_order: 60
+parent: Tools
+grand_parent: Agents and tools
+---
+
+# PPL tool
+**Introduced 2.12**
+{: .label .label-purple }
+
+This is an experimental feature and is not recommended for use in a production environment. For updates on the progress of the feature or if you want to leave feedback, see the associated [GitHub issue](https://github.com/opensearch-project/ml-commons/issues/1161).    
+{: .warning}
+
+The `PPLTool` translates natural language into a PPL query. The tool has a flag to specify whether to run the query. If you set the flag to `true`, the `PPLTool` runs the query and returns the query and the results. 
+
+## Prerequisite
+
+To create a PPL tool, you need a fine-tuned model that translates natural language into PPL queries.
```

Comment:
we also support prompt based translation using large language models like openai and claude. just need to switch the `"model_type": "FINETUNE",` to `"model_type": "CLAUDE",` or `"model_type": "OPENAI",`